src/storage/src/hummock/sstable/builder.rs

```diff
@@ -242,27 +241,39 @@ impl<W: SstableWriter> SstableBuilder<W> {
             } else {
                 vec![]
             },
-            estimated_size: data_len as u32,
+            estimated_size: 0,
             key_count: self.key_count as u32,
             smallest_key,
             largest_key,
             version: VERSION,
+            meta_offset,
+        };
+        meta.estimated_size = meta.encoded_size() as u32 + meta_offset as u32;
+        let sst_info = SstableInfo {
+            id: self.sstable_id,
+            key_range: Some(risingwave_pb::hummock::KeyRange {
+                left: meta.smallest_key.clone(),
+                right: meta.largest_key.clone(),
+                inf: false,
+            }),
+            file_size: meta.estimated_size as u64,
```

Comment:
Now `file_size` will include sst meta size and sst data size. I think `file_size` is heavily used in our compaction strategy. Will this cause any problem (especially for smalle SSTs)?

Reply:
I guess it is not a large problem because the percent of it is very small..

Reply:
A good idea is that we can remove estimate bloom filter size because we can control memory usage of meta-file and data-file by the same buffer

Reply:
I made a mistake. We still need to estimate the size of bloom filter. But I think it is not a problem because since we use prefix bloom filter, the size of bloom filter could be small and we only need to left 1/24 (table_id + key + epoch) of capacity for bloom filter.

Reply:
> I made a mistake. We still need to estimate the size of bloom filter. But I think it is not a problem because since we use prefix bloom filter, the size of bloom filter could be small and we only need to left 1/24 (table_id + key + epoch) of capacity for bloom filter.

ok